File: cap.py
import socket
import threading

def handle_client(client_socket):
    # Connect to the destination server
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.connect(("batman_server.com", 80))

    # Forward data between client and server
    client_thread = threading.Thread(target=forward_data, args=(client_socket, server_socket))
    server_thread = threading.Thread(target=forward_data, args=(server_socket, client_socket))

    client_thread.start()
    server_thread.start()

    client_thread.join()
    server_thread.join()

    client_socket.close()
    server_socket.close()

def forward_data(source, destination):
    data = source.recv(4096)
    while len(data):
        destination.send(data)
        data = source.recv(4096)

def main():
    proxy = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    proxy.bind(("127.0.0.1", 8888))  # Use the desired local port
    proxy.listen(5)

    print("Proxy server started on port 8888...")

    while True:
        client_socket, addr = proxy.accept()
        print(f"Accepted connection from {addr[0]}:{addr[1]}")
        client_handler = threading.Thread(target=handle_client, args=(client_socket,))
        client_handler.start()

if __name__ == "__main__":
    main()

File: cap.java
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.ServerSocket;
import java.net.Socket;

public class cap {
    public static void main(String[] args) {
        try {
            ServerSocket proxySocket = new ServerSocket(8888); // Use the desired local port
            System.out.println("Proxy server started on port 8888...");

            while (true) {
                Socket clientSocket = proxySocket.accept();
                System.out.println("Accepted connection from " + clientSocket.getInetAddress() +
                        ":" + clientSocket.getPort());

                new Thread(() -> handleClient(clientSocket)).start();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static void handleClient(Socket clientSocket) {
        try {
            // Connect to the destination server
            Socket serverSocket = new Socket("batman_server.com", 80);
            System.out.println("Connected to destination server");

            // Forward data between client and server
            Thread clientToServer = new Thread(() -> forwardData(clientSocket, serverSocket));
            Thread serverToClient = new Thread(() -> forwardData(serverSocket, clientSocket));

            clientToServer.start();
            serverToClient.start();

            clientToServer.join();
            serverToClient.join();

            clientSocket.close();
            serverSocket.close();
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }
    }

    private static void forwardData(Socket source, Socket destination) {
        try (InputStream sourceInputStream = source.getInputStream();
            OutputStream destinationOutputStream = destination.getOutputStream()) {

            byte[] buffer = new byte[4096];
            int bytesRead;
            while ((bytesRead = sourceInputStream.read(buffer)) != -1) {
                destinationOutputStream.write(buffer, 0, bytesRead);
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
